modules/katamari/src/test/ts/atomic/api/data/ResultValueTest.ts

```diff
@@ -1,130 +1,133 @@
+      throw new Error('Should not get here!');
+    }, (v) => v + 6), 11);
+
+    assert.deepEqual(s.map((v) => v * 2).getOrDie(), 10);
+
+    assert.deepEqual(s.bind((v) => Result.value('test' + v)).getOrDie(), 'test5');
+
+    assert.deepEqual(s.exists(Fun.always), true);
+    assert.deepEqual(s.forall(Fun.never), false);
+
+    assert.deepEqual(Result.value(5).toOptional().isSome(), true);
+  });
+
+  it('Checking value.is(value.getOrDie()) === true', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      assert.deepEqual(res.is(res.getOrDie()), true);
+    }));
+  });
+
+  it('Checking value.isValue === true', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      assert.deepEqual(res.isValue(), true);
+    }));
+  });
+
+  it('Checking value.isError === false', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      assert.deepEqual(res.isError(), false);
+    }));
+  });
+
+  it('Checking value.getOr(v) === value.value', () => {
+    fc.assert(fc.property(fc.integer(), (a) => {
+      assert.deepEqual(Result.value(a).getOr(a), a);
+    }));
+  });
+
+  it('Checking value.getOrDie() does not throw', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      res.getOrDie();
+    }));
+  });
+
+  it('Checking value.or(oValue) === value', () => {
+    fc.assert(fc.property(fc.integer(), fc.integer(), (a, b) => {
+      assertResult(Result.value(a).or(Result.value(b)), Result.value(a));
+    }));
+  });
+
+  it('Checking error.or(value) === value', () => {
+    fc.assert(fc.property(fc.integer(), fc.integer(), (a, b) => {
+      assertResult(Result.error(a).or(Result.value(b)), Result.value(b));
+    }));
+  });
+
+  it('Checking value.orThunk(die) does not throw', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      res.orThunk(Fun.die('dies'));
+    }));
+  });
+
+  it('Checking value.fold(die, id) === value.getOrDie()', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), (res) => {
+      const actual = res.getOrDie();
+      assert.deepEqual(res.fold(Fun.die('should not get here'), Fun.identity), actual);
+    }));
+  });
+
+  it('Checking value.map(f) === f(value.getOrDie())', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), fc.func(fc.json()), (res, f) => {
+      assert.deepEqual(f(res.getOrDie()), res.map(f).getOrDie());
+    }));
+  });
+
+  it('Checking value.each(f) === undefined', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), fc.func(fc.json()), (res, f) => {
+      const actual = res.each(f);
+      assert.deepEqual(actual, undefined);
+    }));
+  });
+
+  it('Given f :: s -> RV, checking value.bind(f).getOrDie() === f(value.getOrDie()).getOrDie()', () => {
+    fc.assert(fc.property(arbResultValue(fc.integer()), fc.func(arbResultValue(fc.integer())), (res, f) => {
+      assert.deepEqual(f(res.getOrDie()).getOrDie(), res.bind(f).getOrDie());
```

Comment:
I think this one can be `equal` as well, but not 100% sure.

Reply:
Yep.